python312Packages.pycrdt: 0.10.9 -> 0.12.8

GaetanLepage opened the pull request:
## Things done

- **python312Packages.pycrdt: 0.10.9 -> 0.12.8**
  -> Changelog: https://github.com/jupyter-server/pycrdt/blob/0.12.8/CHANGELOG.md
- **python312Packages.pycrdt-websocket: 0.15.1 -> 0.15.4**
  -> Changelog: https://github.com/jupyter-server/pycrdt-websocket/blob/refs/tags/v0.15.4/CHANGELOG.md
- **python312Packages.jupyter-ydoc: 3.0.2 -> 3.0.3**
  -> Changelog: https://github.com/jupyter-server/jupyter_ydoc/blob/v3.0.3/CHANGELOG.md

---

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

GaetanLepage committed:
python312Packages.pycrdt: 0.10.9 -> 0.12.8

Diff:
https://github.com/jupyter-server/pycrdt/compare/refs/tags/0.10.9...0.12.8

Changelog:
https://github.com/jupyter-server/pycrdt/blob/0.12.8/CHANGELOG.md

GaetanLepage committed:
python312Packages.pycrdt-websocket: 0.15.1 -> 0.15.3

Diff: https://github.com/jupyter-server/pycrdt-websocket/compare/refs/tags/v0.15.1...v0.15.3

Changelog: https://github.com/jupyter-server/pycrdt-websocket/blob/refs/tags/v0.15.3/CHANGELOG.md

lucasew commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 370369`

---
### `x86_64-linux`
<details>
  <summary>:x: 18 packages failed to build:</summary>
  <ul>
    <li>python312Packages.jupyter-collaboration</li>
    <li>python312Packages.jupyter-collaboration.dist</li>
    <li>python312Packages.jupyter-server-ydoc</li>
    <li>python312Packages.jupyter-server-ydoc.dist</li>
    <li>python312Packages.jupyter-ydoc</li>
    <li>python312Packages.jupyter-ydoc.dist</li>
    <li>python312Packages.pycrdt-websocket</li>
    <li>python312Packages.pycrdt-websocket.dist</li>
    <li>python313Packages.jupyter-collaboration</li>
    <li>python313Packages.jupyter-collaboration.dist</li>
    <li>python313Packages.jupyter-server-ydoc</li>
    <li>python313Packages.jupyter-server-ydoc.dist</li>
    <li>python313Packages.jupyter-ydoc</li>
    <li>python313Packages.jupyter-ydoc.dist</li>
    <li>python313Packages.pycrdt</li>
    <li>python313Packages.pycrdt-websocket</li>
    <li>python313Packages.pycrdt-websocket.dist</li>
    <li>python313Packages.pycrdt.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python312Packages.pycrdt</li>
    <li>python312Packages.pycrdt.dist</li>
  </ul>
</details>

kirillrdy reviewed (approved):
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 370369`

---
### `x86_64-linux`
<details>
  <summary>:x: 16 packages failed to build:</summary>
  <ul>
    <li>python312Packages.jupyter-collaboration</li>
    <li>python312Packages.jupyter-collaboration.dist</li>
    <li>python312Packages.jupyter-server-ydoc</li>
    <li>python312Packages.jupyter-server-ydoc.dist</li>
    <li>python312Packages.jupyter-ydoc</li>
    <li>python312Packages.jupyter-ydoc.dist</li>
    <li>python313Packages.jupyter-collaboration</li>
    <li>python313Packages.jupyter-collaboration.dist</li>
    <li>python313Packages.jupyter-server-ydoc</li>
    <li>python313Packages.jupyter-server-ydoc.dist</li>
    <li>python313Packages.jupyter-ydoc</li>
    <li>python313Packages.jupyter-ydoc.dist</li>
    <li>python313Packages.pycrdt</li>
    <li>python313Packages.pycrdt-websocket</li>
    <li>python313Packages.pycrdt-websocket.dist</li>
    <li>python313Packages.pycrdt.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.pycrdt</li>
    <li>python312Packages.pycrdt-websocket</li>
    <li>python312Packages.pycrdt-websocket.dist</li>
    <li>python312Packages.pycrdt.dist</li>
  </ul>
</details>

lucasew commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 370369`

---
### `x86_64-linux`
<details>
  <summary>:x: 16 packages failed to build:</summary>
  <ul>
    <li>python312Packages.jupyter-collaboration</li>
    <li>python312Packages.jupyter-collaboration.dist</li>
    <li>python312Packages.jupyter-server-ydoc</li>
    <li>python312Packages.jupyter-server-ydoc.dist</li>
    <li>python312Packages.jupyter-ydoc</li>
    <li>python312Packages.jupyter-ydoc.dist</li>
    <li>python313Packages.jupyter-collaboration</li>
    <li>python313Packages.jupyter-collaboration.dist</li>
    <li>python313Packages.jupyter-server-ydoc</li>
    <li>python313Packages.jupyter-server-ydoc.dist</li>
    <li>python313Packages.jupyter-ydoc</li>
    <li>python313Packages.jupyter-ydoc.dist</li>
    <li>python313Packages.pycrdt</li>
    <li>python313Packages.pycrdt-websocket</li>
    <li>python313Packages.pycrdt-websocket.dist</li>
    <li>python313Packages.pycrdt.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.pycrdt</li>
    <li>python312Packages.pycrdt-websocket</li>
    <li>python312Packages.pycrdt-websocket.dist</li>
    <li>python312Packages.pycrdt.dist</li>
  </ul>
</details>

natsukium commented on pkgs/development/python-modules/pycrdt-websocket/default.nix:
```diff
@@ -26,18 +26,22 @@
 
   build-system = [ hatchling ];
 
+  pythonRelaxDeps = [
+    "pycrdt"
+  ];
```
Unless other packages require a newer version of pycrdt, I don't see the need to upgrade by relaxing the restrictions.
It may be a meaningless upper bound, but these two libraries are tied together and I would like to follow them upstream as much as possible.

GaetanLepage committed:
python312Packages.jupyter-ydoc: 3.0.2 -> 3.0.3

Changelog: https://github.com/jupyter-server/jupyter_ydoc/blob/v3.0.3/CHANGELOG.md
